src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -363,7 +364,15 @@ export class SavedObjectsRepository {
       ...(Array.isArray(references) && { references }),
     });
 
-    const raw = this._serializer.savedObjectToRaw(migrated as SavedObjectSanitizedDoc);
+    /**
+     * If a validation has been registered for this type, we run it against the migrated attributes.
+     * This is an imperfect solution because malformed attributes could have already caused the
+     * migration to fail, but it's the best we can do without devising a way to run validations
+     * inside the migration algorithm itself.
+     */
```

Comment:
(thinking out loud) After thinking about it again and looking at the doc migrator's code, I wonder if going so would have been that hard in the end. But the implementation is what we agreed on, so it's fine.